src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,122 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
     @Override
     public void put(K key, V value) {
-
+        if (table == null || size == 0) {
+            resize();
+        }
+        int index = hash(key) % table.length;
+        if (table[index] != null) {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (currentNode.key == key
+                        || (currentNode.key != null && currentNode.key.equals(key))) {
+                    currentNode.value = value;
+                    break;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(hash(key), key, value, null);
+                    size++;
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        } else {
+            table[index] = new Node<>(hash(key), key, value, null);
+            size++;
+        }
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
+            int index = hash(key) % table.length;
+            if (table[index] != null) {
```

Comment:
it seems that if you remove this "if" statement nothing will change.
if table[index] is null then expression Node<K, V> currentNode = table[index]; will make currentNode == null
and while (currentNode != null) { won't be executed